File: Guanabara - Java/Meus exercícios POO/Aula10/src/main/java/heranca/Professor.java
package heranca;
public class Professor extends Pessoa { //herança
    //Atributos
    private String especialidade;
    private float salario;
    
    //Método Público
    public void receberAumento() {
        
    }
    //Métodos Acessores
    public String getEspecialidade() {
        return especialidade;
    }

    public void setEspecialidade(String especialidade) {
        this.especialidade = especialidade;
    }

    public float getSalario() {
        return salario;
    }

    public void setSalario(float salario) {
        this.salario = salario;
    }
}

File: PHP/exercicios-PHPOO/Aula10-HerancaPessoa/Professor.php
<?php
include_once "Pessoa.php";
class Professor extends Pessoa
{
    //Atributos
    private $especialidade;
    private $salario;

    //Métodos Especias
    function receberAumento($quantia)
    {
        $salarioAnterior = $this->salario;
        $this->setSalario($this->getSalario() + $quantia);
        $porcentagem = (intdiv(($this->getSalario() * 100), $salarioAnterior) - 100);
        echo "<p>O salário do professor <strong>" . $this->nome . "</strong> recebeu um aumento de <strong>" . $porcentagem . "%</strong>. Seu novo salário agora é <strong>R$" . $this->salario . "</strong></p>";
    }

    //Métodos Acessores
    function getEspecialidade()
    {
        return $this->especialidade;
    }

    function getSalario()
    {
        return $this->salario;
    }
    function setEspecialidade($Especialidade)
    {
        $this->especialidade = $Especialidade;
    }
    function setSalario($Salario)
    {
        $this->salario = $Salario;
    }
}
